### PR TITLE
Update ViewF Semigroupal implementation and check for modCB Eq in tests

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -10,6 +14,17 @@ lazy val root = tlCrossRootProject.aggregate(crystal)
 lazy val crystal = crossProject(JVMPlatform, JSPlatform)
   .in(file("."))
   .settings(
+    mimaBinaryIssueFilters ++= Seq(
```
Comment (rpiaggio) [inline build.sbt:17]: Why is this necessary?

Comment (toddburnside) [reply build.sbt:17]: It seems to me it might be better to just bump the baseVersion.

Comment (hugo-vrijswijk) [reply build.sbt:17]: Also fine by me. I added it because the `Semigroup` and `Invariant` were two separate given values, now combined. But mima sees them as "removed". I can either remove this line after the next release (I think) or bump the baseVersion

**Review by rpiaggio — APPROVED** [review]
LGTM
